### PR TITLE
[rrd4j] improvements and extensions to documentation

### PR DESCRIPTION
**Pull request by andrewfg**
When I tried to set up rrd4j persistence on my own system, I found the `rrd4j` `ReadMe.md` documentation in general very hard to understand, and in particular missing some specific information.

Very kindly @JueBag helped me to solve it -- thank you!

So, based on our common learning, this PR contains improvements and extensions to the `ReadMe.md` file (it also contains a small change in the `.cfg` sample file in which the parameter names have been changed to match the documentation).

### CODE REVIEW

**Comment by andrewfg** [comment]
@JueBag this fails the DCO because your commit sha: be0820a reported `Expected "Jürgen Baginski opus42@gmx.de", but got "Jürgen Baginski opus42@gmx.de"` -- I am not sure what that means, or how to fix it (???)

**Comment by JueBag** [comment]
Again a problem with the SignOff, I have no idea why and how to fix it. Sorry for that.

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md [hunk 1]
```diff
@@ -1,123 +1,183 @@
...
 
-<!-- /MarkdownTOC -->
+The persistence service reads data "samples" from the OpenHAB core at regular intervals, and these are then put into the storage boxes.
+Either a) the samples are stored singly directly into a box, or b) multiple samples are consolidated (using a consolidation function) into a box.
+
+The service starts by storing samples in the leftmost box in the drawer.
+Once the leftmost box is full, the service starts filling the next box to the right; and so on.
+Once the rightmost box in the drawer is full, the leftmost box is emptied, the content of all boxes is moved one box to the left, and new content is added to the rightmost box.
+
+So for example..
```
Comment (9037568) [inline bundles/persistence/org.openhab.persistence.rrd4j/README.md:28]: I advocate for expanding this line to indicate that the example contains values that can vary, and then remove the "(say)" parentheticals from the lines below.

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md [hunk 2]
```diff
@@ -1,123 +1,183 @@
...
+
+- Samples are taken at intervals of (say) `60` seconds
+- They are consolidated by (say) the `AVERAGE` function, over (say) `10` samples, into boxes i.e. a box covers `10 X 60` seconds
+- The ful archive contains (say) `250` boxes i.e. the archive/drawer covers `60 X 10 X 250` seconds
```
Comment (9037568) [inline bundles/persistence/org.openhab.persistence.rrd4j/README.md:32]: "full archive"

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md [hunk 3]
```diff
@@ -1,123 +1,183 @@
...
+### `<sampleInterval>` (Sample Interval)
+
+The time interval (seconds) between reading consecutive samples from the OpenHAB core.
+
+It must be a positive integer value.
+
+### `<consolidationFunction>` (Consolidation Function)
+
+Determines the type of data compression to be used when more than one sample is to be stored in a single "storage box".
+So if you use (say) the `AVERAGE` function, and two samples of `20.0` and `21.0` are to be stored, then the value `20.5` would be stored in the box.
```
Comment (9037568) [inline bundles/persistence/org.openhab.persistence.rrd4j/README.md:111]: <s>(say)</s>

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md [hunk 4]
```diff
@@ -1,123 +1,183 @@
...
 
-Each datasource also has a value for heartbeat, minimum and maximum. This heartbeat setting helps the database to detect "missing" values, i.e. if no new value is stored after "heartbeat" seconds, the value is considered missing when charting. Minimum and maximum define the range of acceptable values for that datasource.
+It must be one of the following strings:
 
-### Step(s)
+- **AVERAGE** the average of all the samples is stored in the box
+- **MIN** the highest sample is stored in the box
```
Comment (9037568) [inline bundles/persistence/org.openhab.persistence.rrd4j/README.md:116]: Don't you have MIN and MAX reversed here?

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md (hunk 1 above)
Comment (andrewfg) [reply bundles/persistence/org.openhab.persistence.rrd4j/README.md:28]: Good idea, thank you.

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md (hunk 2 above)
Comment (andrewfg) [reply bundles/persistence/org.openhab.persistence.rrd4j/README.md:32]: Ok

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md (hunk 3 above)
Comment (andrewfg) [reply bundles/persistence/org.openhab.persistence.rrd4j/README.md:111]: Ok

File: bundles/persistence/org.openhab.persistence.rrd4j/README.md (hunk 4 above)
Comment (andrewfg) [reply bundles/persistence/org.openhab.persistence.rrd4j/README.md:116]: Oops!

**Comment by andrewfg** [comment]
@9037568 I have pushed your proposed changes. Do you know why the DCO test keeps failing?

**Comment by 9037568** [comment]
I haven't the first clue about the DCO. The values it's claiming don't match look like they match to me.
Thanks for the changes.

**Comment by andrewfg** [comment]
@9037568 in the meantime, it seems that all tests have passed. I don’t know why it errored before, but anyway it is “all green now”.

**Comment by andrewfg** [comment]
Ping!

IMHO this PR is ready to be published. Note: it is documentation only..

**Review by kaikreuzer — APPROVED** [review]
Absolutely, thanks for the ping.

**Comment by kaikreuzer** [comment]
@andrewfg May I ask you to create another PR to make the same update available in openHAB 3 at https://github.com/openhab/openhab-addons/tree/master/bundles/org.openhab.persistence.rrd4j? Thanks!

**Comment by andrewfg** [comment]
> @andrewfg May I ask you to create another PR to make the same update available in openHAB 3 at https://github.com/openhab/openhab-addons/tree/master/bundles/org.openhab.persistence.rrd4j? Thanks!

@kaikreuzer ok, done.

**Comment by kaikreuzer** [comment]
Thanks!

**Pull request merged**